lib/livebook/stamping.ex

```diff
@@ -37,14 +37,15 @@ defmodule Livebook.Stamping do
     end
   end
 
-  defp derive_keys(secret_key) do
+  @doc """
+  Derives the secret and sign secret from given `secret_key`.
+  """
+  @spec derive_keys(String.t(), String.t()) :: {bitstring(), bitstring()}
+  def derive_keys(secret_key, topic) do
```

Comment:
I would keep this specialized to stamping, so this module stays focused. We can have a similar private function in `Teams` alongside secret encrypt/decrypt with its own salt, it's almost a single function call anyway :)